staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -472,20 +498,15 @@ func (g *Cloud) getInstancesByNames(names []string) ([]*gceInstance, error) {
 		}
 	}
 
-	if remaining > 0 {
-		var failed []string
-		for k := range found {
-			if found[k] == nil {
-				failed = append(failed, k)
-			}
-		}
-		klog.Errorf("Failed to retrieve instances: %v", failed)
-		return nil, cloudprovider.InstanceNotFound
+	if len(extraInstancesFound) > 0 {
```

Comment:
what is the purpose of this log?

Reply:
its a warning that were making more calls to the API than necessary

Reply:
the API to list instances (with prefix if available) will most likely have more instances then requested. Why we need a warning here?

Reply:
It shouldnt if the instance group is set up properly i think - I can remove it if its such a concern. I figured its better to get a warning that youre potentially making more requests than you need to than to have less logging.

Reply:
the caller will try to list the nodes that have volumes attached to it, not all nodes might have volumes attached

Reply:
I meant listing extra instances for the function itself, independent of what the caller does with it. So if the prefix is removed, then its possible to list all instances in your project instead of just the ones in your cluster.

Reply:
I just removed this because its not important.